File: ruby/3a.rb
#!/usr/bin/ruby

# Convert input into an array of arrays of side lengths
input = File.open('3.in').readlines.map{|x| x.split.map(&:to_i)}

# Select the triangles whose border length less the maximum is greater than the maximum
puts input.select{|t| t.reduce(:+) - t.max > t.max}.size

File: python/3a.py
#!/usr/bin/python

def main():
    # Process input into a list of lists for side lengths, ordered
    data = [sorted((int(y) for y in x.split()), reverse=True) for x in open('3.in')]

    # Select triangles whose border less the largest side is larger than the largest side
    valid = [x for x in data if sum(x) - x[0] > x[0]]

    # Answer is the number of valid triangles
    print len(valid)

if __name__ == '__main__':
    main()

# vi: ts=4 et:
